### ISSUE TITLE
Can't delete rc when the pods' AGE is invalid

### ISSUE THREAD

**Issue by JunejaTung** [kubernetes/kubernetes#43211]
**Lock when delete rc**
```
the kubectl hook on deleting replicationcontroller
[root@wlan-cloudserver31 temp]# kubectl  delete -f webserver-controller.yaml 
error: error when stopping "webserver-controller.yaml": timed out waiting for the condition
[root@wlan-cloudserver31 temp]# 
```

**Steps to reproduce the issue:**
1.set datetime 8 hours later
```
 [root@wlan-cloudserver31 home]# date 03162317
 Thu Mar 16 23:17:00 CST 2017
```
2. create pods
```
root@wlan-cloudserver31 temp]# kubectl create -f webserver-controller.yaml 
replicationcontroller "webserver" created
[root@wlan-cloudserver31 temp]# kubectl  get pods -o wide
NAME              READY     STATUS    RESTARTS   AGE       NODE
webserver-04mp0   1/1       Running   0          10s       172.34.37.1
```
3. set datetime to current right 
```
[root@wlan-cloudserver31 temp]# date
Thu Mar 16 23:19:26 CST 2017
[root@wlan-cloudserver31 temp]# date 03161620
Thu Mar 16 16:20:00 CST 2017
[root@wlan-cloudserver31 temp]# kubectl  get pods -o wide
NAME              READY     STATUS    RESTARTS   AGE         NODE
webserver-04mp0   1/1       Running   0          <invalid>   172.34.37.1
```

**Describe the results you received:**
it hook on delete rc
```
[root@wlan-cloudserver31 temp]# kubectl  delete -f webserver-controller.yaml 
error: error when stopping "webserver-controller.yaml": timed out waiting for the condition
[root@wlan-cloudserver31 wlanuser]# kubectl delete  rc webserver
error: timed out waiting for the condition
```
set datetime to 8 hours later , AGE is valid, failed to delete as before 
```
[root@wlan-cloudserver31 wlanuser]# date
Thu Mar 16 16:50:50 CST 2017
[root@wlan-cloudserver31 wlanuser]# date 03170049
Fri Mar 17 00:49:00 CST 2017
[root@wlan-cloudserver31 wlanuser]# kubectl get pods 
NAME              READY     STATUS    RESTARTS   AGE
webserver-04mp0   1/1       Running   0          1h
[root@wlan-cloudserver31 wlanuser]# kubectl delete rc webserver
error: timed out waiting for the condition
[root@wlan-cloudserver31 wlanuser]# 
```
**Describe the results you expected:**
```
[root@wlan-cloudserver31 wlanuser]# kubectl delete rc webserver
replicationcontroller "webserver" deleted
```
**rc yaml, volumes:**
````yaml
            value: "8080"
        volumeMounts:
          - name: syslog
            mountPath: /workspace/logs
          - name: optiot
            mountPath: /opt/iot
          - name: systime
            readOnly: true
            mountPath: /etc/localtime
      volumes:
        - name: syslog
          hostPath: {path: /opt/iotlog/servers}
        - name: optiot
          hostPath: {path: /opt/iot}
        - name: systime
          hostPath: {path: /etc/localtime}
````

**Strace results:**
```
[root@wlan-cloudserver31 home]# strace -e trace=all -p 117841
Process 117841 attached
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
epoll_wait(6, {}, 128, 0)               = 0
epoll_wait(6, {{EPOLLIN|EPOLLOUT, {u32=2798273456, u64=140684452053936}}}, 128, -1) = 1
futex(0x3422cb0, FUTEX_WAKE, 1)         = 1
read(5, "HTTP/1.1 200 OK\r\nContent-Type: a"..., 4096) = 2281
futex(0xc8202da108, FUTEX_WAKE, 1)      = 1
futex(0xc820474508, FUTEX_WAKE, 1)      = 1
futex(0xc820474508, FUTEX_WAKE, 1)      = 1
read(5, 0xc82012f000, 4096)             = -1 EAGAIN (Resource temporarily unavailable)
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x3422cc8, FUTEX_WAKE, 1)         = 0
futex(0x3422bf0, FUTEX_WAKE, 1)         = 1
futex(0xc820062508, FUTEX_WAKE, 1)      = 1
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
sched_yield()                           = 0
sched_yield()                           = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
sched_yield()                           = 0
futex(0x3423090, FUTEX_WAIT, 2, NULL)   = 0
futex(0x3423090, FUTEX_WAKE, 1)         = 0
futex(0x3422cc8, FUTEX_WAIT, 0, {0, 100000}) = -1 EAGAIN (Resource temporarily unavailable)
futex(0x3422ca0, FUTEX_WAKE, 1)         = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = -1 EAGAIN (Resource temporarily unavailable)
sched_yield()                           = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x3422ca0, FUTEX_WAKE, 1)         = 1
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = -1 EAGAIN (Resource temporarily unavailable)
sched_yield()                           = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
sched_yield()                           = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
epoll_wait(6, {}, 128, 0)               = 0
epoll_wait(6, {{EPOLLIN|EPOLLOUT, {u32=2798273456, u64=140684452053936}}}, 128, -1) = 1
futex(0x3422cb0, FUTEX_WAKE, 1)         = 1
read(5, "HTTP/1.1 200 OK\r\nContent-Type: a"..., 4096) = 2281
futex(0xc8202da108, FUTEX_WAKE, 1)      = 1
read(5, 0xc82012f000, 4096)             = -1 EAGAIN (Resource temporarily unavailable)
epoll_wait(6, {}, 128, 0)               = 0
epoll_wait(6, {{EPOLLIN|EPOLLOUT, {u32=2798273456, u64=140684452053936}}}, 128, -1) = 1
futex(0x3422cb0, FUTEX_WAKE, 1)         = 1
read(5, "HTTP/1.1 200 OK\r\nContent-Type: a"..., 4096) = 2281
futex(0xc8202da108, FUTEX_WAKE, 1)      = 1
read(5, 0xc82012f000, 4096)             = -1 EAGAIN (Resource temporarily unavailable)
epoll_wait(6, {}, 128, 0)               = 0
epoll_wait(6, {{EPOLLIN|EPOLLOUT, {u32=2798273456, u64=140684452053936}}}, 128, -1) = 1
futex(0x3422cb0, FUTEX_WAKE, 1)         = 1
read(5, "HTTP/1.1 200 OK\r\nContent-Type: a"..., 4096) = 2281
futex(0xc820474508, FUTEX_WAKE, 1)      = 1
futex(0xc8202da108, FUTEX_WAKE, 1)      = 1
read(5, 0xc82012f000, 4096)             = -1 EAGAIN (Resource temporarily unavailable)
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
epoll_wait(6, {}, 128, 0)               = 0
epoll_wait(6, {{EPOLLIN|EPOLLOUT, {u32=2798273456, u64=140684452053936}}}, 128, -1) = 1
futex(0x3422cb0, FUTEX_WAKE, 1)         = 1
read(5, "HTTP/1.1 200 OK\r\nContent-Type: a"..., 4096) = 2281
futex(0xc820474508, FUTEX_WAKE, 1)      = 1
read(5, 0xc82012f000, 4096)             = -1 EAGAIN (Resource temporarily unavailable)
epoll_wait(6, {}, 128, 0)               = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
epoll_wait(6, {}, 128, 0)               = 0
epoll_wait(6, {{EPOLLIN|EPOLLOUT, {u32=2798273456, u64=140684452053936}}}, 128, -1) = 1
futex(0x3422cb0, FUTEX_WAKE, 1)         = 1
read(5, "HTTP/1.1 200 OK\r\nContent-Type: a"..., 4096) = 2281
futex(0xc8202da108, FUTEX_WAKE, 1)      = 1
futex(0xc820474508, FUTEX_WAKE, 1)      = 1
futex(0xc820474508, FUTEX_WAKE, 1)      = 1
read(5, 0xc82012f000, 4096)             = -1 EAGAIN (Resource temporarily unavailable)
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
epoll_wait(6, {}, 128, 0)               = 0
epoll_wait(6, {{EPOLLIN|EPOLLOUT, {u32=2798273456, u64=140684452053936}}}, 128, -1) = 1
futex(0x3422cb0, FUTEX_WAKE, 1)         = 1
read(5, "HTTP/1.1 200 OK\r\nContent-Type: a"..., 4096) = 2281
futex(0xc8202da108, FUTEX_WAKE, 1)      = 1
futex(0xc820474508, FUTEX_WAKE, 1)      = 1
read(5, 0xc82012f000, 4096)             = -1 EAGAIN (Resource temporarily unavailable)
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0xc820474508, FUTEX_WAKE, 1)      = 1
futex(0x3422238, FUTEX_WAIT, 0, {0, 973523821}) = -1 ETIMEDOUT (Connection timed out)
futex(0x3422cb0, FUTEX_WAKE, 1)         = 1
futex(0xc8202da108, FUTEX_WAKE, 1)      = 1
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
epoll_wait(6, {}, 128, 0)               = 0
epoll_wait(6, {{EPOLLIN|EPOLLOUT, {u32=2798273456, u64=140684452053936}}}, 128, -1) = 1
futex(0x3422cb0, FUTEX_WAKE, 1)         = 1
read(5, "HTTP/1.1 200 OK\r\nContent-Type: a"..., 4096) = 2281
futex(0xc820063108, FUTEX_WAKE, 1)      = 1
read(5, 0xc82012f000, 4096)             = -1 EAGAIN (Resource temporarily unavailable)
epoll_wait(6, {}, 128, 0)               = 0
epoll_wait(6, {{EPOLLIN|EPOLLOUT, {u32=2798273456, u64=140684452053936}}}, 128, -1) = 1
futex(0x3422cb0, FUTEX_WAKE, 1)         = 1
read(5, "HTTP/1.1 200 OK\r\nContent-Type: a"..., 4096) = 2281
futex(0xc820062508, FUTEX_WAKE, 1)      = 1
futex(0xc820474508, FUTEX_WAKE, 1)      = 1
read(5, 0xc82012f000, 4096)             = -1 EAGAIN (Resource temporarily unavailable)
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
epoll_wait(6, {}, 128, 0)               = 0
select(0, NULL, NULL, NULL, {0, 100})   = 0 (Timeout)
epoll_wait(6, {{EPOLLIN|EPOLLOUT, {u32=2798273456, u64=140684452053936}}}, 128, -1) = 1
futex(0x3422cb0, FUTEX_WAKE, 1)         = 1
read(5, "HTTP/1.1 200 OK\r\nContent-Type: a"..., 4096) = 2281
futex(0xc820062508, FUTEX_WAKE, 1)      = 1
read(5, 0xc82012f000, 4096)             = -1 EAGAIN (Resource temporarily unavailable)
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
select(0, NULL, NULL, NULL, {0, 100})   = 0 (Timeout)
futex(0x34239c8, FUTEX_WAIT, 0, NULL)   = 0
epoll_wait(6, {}, 128, 0)               = 0
select(0, NULL, NULL, NULL, {0, 100})   = 0 (Timeout)
epoll_wait(6, {{EPOLLIN|EPOLLOUT, {u32=2798273456, u64=140684452053936
```

**Output of `docker version`:**

```
webserver-04mp0   1/1       Running   0          <invalid>
[root@wlan-cloudserver31 wlanuser]# kubectl version 
Client Version: version.Info{Major:"1", Minor:"2", GitVersion:"v1.2.0", GitCommit:"ec7364b6e3b155e78086018aa644057edbe196e5", GitTreeState:"clean"}
Server Version: version.Info{Major:"1", Minor:"2", GitVersion:"v1.2.0", GitCommit:"ec7364b6e3b155e78086018aa644057edbe196e5", GitTreeState:"clean"}
[root@wlan-cloudserver31 wlanuser]# docker version 
Client:
 Version:         1.9.1
 API version:     1.21
 Package version: docker-common-1.9.1-40.el7.centos.x86_64
 Go version:      go1.4.2
 Git commit:      ab77bde/1.9.1
 Built:           
 OS/Arch:         linux/amd64

Server:
 Version:         1.9.1
 API version:     1.21
 Package version: docker-common-1.9.1-40.el7.centos.x86_64
 Go version:      go1.4.2
 Git commit:      ab77bde/1.9.1
 Built:           
 OS/Arch:         linux/amd64
```

**Output of `docker info`:**

```
[root@wlan-cloudserver31 wlanuser]# docker info
Containers: 12
Images: 73
Server Version: 1.9.1
Storage Driver: devicemapper
 Pool Name: docker-253:0-138259636-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 107.4 GB
 Backing Filesystem: 
 Data file: /dev/loop10
 Metadata file: /dev/loop11
 Data Space Used: 1.066 GB
 Data Space Total: 107.4 GB
 Data Space Available: 23.01 GB
 Metadata Space Used: 2.404 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.145 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /opt/iotdata/docker/devicemapper/devicemapper/data
 Metadata loop file: /opt/iotdata/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.135-RHEL7 (2016-09-28)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.10.0-229.el7.x86_64
Operating System: CentOS Linux 7 (Core)
CPUs: 8
Total Memory: 15.5 GiB
Name: wlan-cloudserver31
ID: SJFP:O4HP:2TEP:GWRT:WLKT:PCVC:UJLE:YUG5:EQCK:X6SD:6TOC:HCG2
WARNING: bridge-nf-call-ip6tables is disabled
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by dixudx** [comment]
I can't reproduce this. It has nothing to do with `Pod`'s age.

**Comment by JunejaTung** [comment]
@dixudx   had you mount the host's system datetime to pods in your replicationController yaml?
```
        volumeMounts:
          - name: systime
            readOnly: true
            mountPath: /etc/localtime
      volumes:
        - name: systime
          hostPath: {path: /etc/localtime}
```
then change the host's datetime after the pods running

**Comment by kargakis** [comment]
This seems like a bug in kubectl. @JunejaTung can you try using a newer version of kubectl? Also from 1.4, server-side deletion is supported, although kubectl still uses the client-side reaper for backwards compatibility

**Comment by dixudx** [comment]
@JunejaTung I still cannot reproduce this even with volume `systime` mounted.

Below is my detailed log.

```
[root@xxxxx home]# date
Mon Mar 20 15:18:34 CST 2017
[root@xxxxx home]# cat redis.yaml 
apiVersion: v1
kind: ReplicationController
metadata:
  name: redis
spec:
  replicas: 1
  selector:
    name: redis
  template:
    metadata:
      labels:
        name: redis
    spec:
      containers:
      - name: redis
        image: redis:3
        ports:
        - containerPort: 6379
        volumeMounts:
          - name: systime
            readOnly: true
            mountPath: /etc/localtime
      volumes:
        - name: systime
          hostPath: {path: /etc/localtime}
[root@xxxxx home]# kubectl --kubeconfig=./config create -f redis.yaml 
replicationcontroller "redis" created
[root@xxxxx home]# kubectl --kubeconfig=./config get pods
NAME                       READY     STATUS              RESTARTS   AGE
busybox                    1/1       Running             166        6d
busybox-198                1/1       Running             72         3d
busybox-199                1/1       Running             69         3d
busybox-201                1/1       Running             74         3d
busybox-202                1/1       Running             73         3d
grpc-client-2wj3q          1/1       Running             0          17d
grpc-client-rjsrz          1/1       Running             0          7d
grpc-client-vt72b          1/1       Running             1          6d
grpc-server-t96xn          1/1       Running             0          7d
grpc-server-v2vg0          1/1       Running             0          17d
new1-703568439-nzsw1       1/1       Running             0          7d
redis-1dgcb                0/1       ContainerCreating   0          <invalid>
testapp-2016160035-nck91   1/1       Running             0          12d
[root@xxxxx home]# kubectl --kubeconfig=./config get pods
NAME                       READY     STATUS    RESTARTS   AGE
busybox                    1/1       Running   166        6d
busybox-198                1/1       Running   72         3d
busybox-199                1/1       Running   69         3d
busybox-201                1/1       Running   74         3d
busybox-202                1/1       Running   73         3d
grpc-client-2wj3q          1/1       Running   0          17d
grpc-client-rjsrz          1/1       Running   0          7d
grpc-client-vt72b          1/1       Running   1          6d
grpc-server-t96xn          1/1       Running   0          7d
grpc-server-v2vg0          1/1       Running   0          17d
new1-703568439-nzsw1       1/1       Running   0          7d
redis-1dgcb                1/1       Running   0          33s
testapp-2016160035-nck91   1/1       Running   0          12d
[root@xxxxx home]# date
Mon Mar 20 15:21:19 CST 2017
[root@xxxxx home]# date 03200721
Mon Mar 20 07:21:00 CST 2017
[root@xxxxx home]# kubectl --kubeconfig=./config get pods
NAME                       READY     STATUS    RESTARTS   AGE
busybox                    1/1       Running   166        6d
busybox-198                1/1       Running   72         2d
busybox-199                1/1       Running   69         2d
busybox-201                1/1       Running   74         2d
busybox-202                1/1       Running   73         2d
grpc-client-2wj3q          1/1       Running   0          17d
grpc-client-rjsrz          1/1       Running   0          7d
grpc-client-vt72b          1/1       Running   1          5d
grpc-server-t96xn          1/1       Running   0          7d
grpc-server-v2vg0          1/1       Running   0          17d
new1-703568439-nzsw1       1/1       Running   0          7d
redis-1dgcb                1/1       Running   0          <invalid>
testapp-2016160035-nck91   1/1       Running   0          11d
[root@xxxxx home]# kubectl --kubeconfig=./config delete -f redis.yaml 
replicationcontroller "redis" deleted
[root@xxxxx home]# kubectl --kubeconfig=./config get pods
NAME                       READY     STATUS    RESTARTS   AGE
busybox                    1/1       Running   166        6d
busybox-198                1/1       Running   72         2d
busybox-199                1/1       Running   69         2d
busybox-201                1/1       Running   74         2d
busybox-202                1/1       Running   73         2d
grpc-client-2wj3q          1/1       Running   0          17d
grpc-client-rjsrz          1/1       Running   0          7d
grpc-client-vt72b          1/1       Running   1          5d
grpc-server-t96xn          1/1       Running   0          7d
grpc-server-v2vg0          1/1       Running   0          17d
new1-703568439-nzsw1       1/1       Running   0          7d
testapp-2016160035-nck91   1/1       Running   0          11d
```

**Comment by JunejaTung** [comment]
@dixudx  My kubernetes version is 1.2.0, maybe this bug had been resolved in  1.4.0 as @kargakis  stated. we will try a new version to avoid this problem. thanks

**Comment by randunel** [comment]
This issue persists with 

```
Client Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.2", GitCommit:"08e099554f3c31f6e6f07b448ab3ed78d0520507", GitTreeState:"clean", BuildDate:"2017-01-12T04:57:25Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.6", GitCommit:"114f8911f9597be669a747ab72787e0bd74c9359", GitTreeState:"clean", BuildDate:"2017-03-28T13:36:31Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by JunejaTung** [comment]
had this bug resolved ? I have upgrade kubernetes to version 1.5.x, this bug is still existent。
@xiangpengzhao @kargakis

**Comment by ShualLiu** [comment]
@JunejaTung My environment the  v1.6 has fixed the problem. I will recheck it later.
